addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -375,3 +379,275 @@ def _l10n_es_edi_facturae_render_facturae(self):
         except ValueError:
             errors.append(_('No valid certificate found for this company, Facturae EDI file will not be signed.\n'))
         return xml_content, errors
+
+    # -------------------------------------------------------------------------
+    # IMPORT
+    # -------------------------------------------------------------------------
+
+    def _get_edi_decoder(self, file_data, new=False):
+        def is_facturae(tree):
+            return tree.tag == '{http://www.facturae.es/Facturae/2014/v3.2.1/Facturae}Facturae'
+
+        if file_data['type'] == 'xml' and is_facturae(file_data['xml_tree']):
+            return self._import_invoice_facturae
+
+        return super()._get_edi_decoder(file_data, new=new)
+
+    def _import_invoice_facturae(self, invoice, file_data, new=False):
+        tree = file_data['xml_tree']
+
+        signature_node = tree.find('ds:Signature', namespaces={"ds": "http://www.w3.org/2000/09/xmldsig#"})
+        xades_error_message = None
+
+        if signature_node is not None:
+            try:
+                validate_xades_signature(signature_node)
+            except Exception as e:
+                xades_error_message = str(e)
```

Comment:
```suggestion
        if signature_node is not None:
            try:
                validate_xades_signature(signature_node)
            except Exception as e:
                xades_error_message = str(e)
```
Why silencing the error? is it a PO request? This is not okay
Too Broad exception, please specify the excpected one if we keep that

Reply:
It's not an error, it's so that the user knows why his xml failed the validation. If it was a reference issue, or it wasn't signed in the first place. This is not a PO request, I thought it would be useful though. @chklop What do you think? Should we just show a "Warning: could not validate the xml signature" without specification or tell the reason it failed?

Reply:
```suggestion
        if signature_node is not None:
            try:
                validate_xades_signature(signature_node)
            except (...) as e:
                raise UserError(_("..."))
```
First and foremost, the traceback would be in the logs and if we ever want to have it (99% sure we do not), I would recommend putting it in the UserError or something

Reply:
I don't get the suggestion above, it's the same code ? Also what does this imply, and what's the suggested change ? I think XML signature failing can be anything and will most likely not be of the same magnitude.

Reply:
> 



> Also what does this imply, and what's the suggested change ?

Right now, what I do is if the validation fails, I add a message in the invoice logs saying that it failed and why it failed. For example: references have the wrong digest, or the xml isn't signed in the first place, etc. Julien is suggesting that this is not info the user needs, which I am fine with. Need your PO opinion :smile: 



> If it fails, you don't go further

I do go further. I continue the import as I normally would, I just add a message in the invoice logs saying that the validation failed.

Reply:
Yes, I had a discussion with @chklop and he said to continue the import and just add a message that the signature was not validated.